### ISSUE TITLE
Published type definitions are broken (bad folder structure)

### ISSUE THREAD

**Issue by Jessidhia** [thebuilder/react-intersection-observer#177]
Your output is using a different folder structure than your input, so the published types are broken. `InView.d.ts` and `useInView.d.ts` should be co-located with the `.js` files instead of moved into the `typings` subfolder.

```ts
../node_modules/react-intersection-observer/dist/typings/InView.d.ts:2:43 - error TS2307: Cannot find module './typings/types'.

2 import { IntersectionObserverProps } from './typings/types';
                                            ~~~~~~~~~~~~~~~~~

../node_modules/react-intersection-observer/dist/typings/useInView.d.ts:1:51 - error TS2307: Cannot find module './typings/types'.

1 import { HookResponse, IntersectionOptions } from './typings/types';
                                                    ~~~~~~~~~~~~~~~~~
```

**Comment by thebuilder** [comment]
Thanks for the heads up, not sure how I haven't run into this issue.

The problem is I'm not copying over the the `d.ts` files from the `src/typings` dir.
I'll include them, and that should fix it.

**Comment by Jessidhia** [comment]
You probably use `--skipLibCheck` which is a terrible idea and should never be used 😇 ...but you'll find it in guides and boilerplates everywhere 😠

**Comment by thebuilder** [comment]
I don't think so, but i'll keep that in mind. 👍

**Comment by thebuilder** [comment]
This should be fixed `8.0.3`
